### ISSUE TITLE
Manubot and privat repository

### ISSUE THREAD

**Issue by sebastiansukstorf** [manubot/rootstock#407]
Hello, can I use Manubot with my lector in a privat repository?

**Comment by agitter** [comment]
Yes, we have a few examples of users running Manubot in a private GitHub repository.  There is some recent discussion in #405.

Two modes include:
- Keep the repository source private and do not enable GitHub Pages.  Access the generated `manuscript.pdf` from the `output` branch of the repository.
- Keep the repository source private but enable GitHub Pages to make the generated manuscript public.

**Comment by sebastiansukstorf** [comment]
Hello Anthony, thank you for your answer. I am not quite sure how to use the setup description, correctly. Do I follow all steps for using manubot? Is it possible to get some help from any community member? Thank you very much. Sebastian

**Comment by agitter** [comment]
I can try to help if you get stuck on some of the setup steps.  How far were you able to make it?  With a private manuscript you will follow almost all of the typical setup steps listed at https://github.com/manubot/rootstock/blob/main/SETUP.md#creating-a-new-manuscript:
- Configuration: follow standard instructions
- Create repository: follow standard instructions
- Continuous integration: here you can follow the steps for GitHub Actions, which is the recommended continuous integration service. After you complete the setup steps, visit the settings in your new GitHub repository.  In the GitHub Pages section there is a Source option.  Set this to None to disable GitHub Pages for your private repository.  Then, your manuscript source files and the generated HTML and PDF outputs will be private.
- README updates: follow standard instructions
- Finalize: follow standard instructions

**Comment by sebastiansukstorf** [comment]
Hello Anthony,

My last steps were:
# remove Travis CI config
git rm .travis.yml
# remove AppVeyor config
git rm .appveyor.yml
# remove ci/install.sh if using neither Travis CI nor AppVeyor
git rm ci/install.sh
The installation for SSH, Travis CI and APPVeyor Setup did confuse me.

I receveived in teh Action panel the following error message: Process completed with exit code 83.

THY
Sebastian

**Comment by agitter** [comment]
Does the error message look like this one? https://github.com/fmsabatini/sPlotOpen_Manuscript/actions/runs/354512839  Or is there more printed along with the error message?

That error is caused by using an older version of Manubot rootstock, which did not pin a required package (#387).  However, if you just performed all of the setup steps now, it would be surprising if you do not have the latest version of Manubot rootstock.

**Comment by sebastiansukstorf** [comment]
Hello Anthony,

I followed the procedure until step: git push --set-upstream origin master

I received the following error message in my terminal:

**Comment by dhimmel** [comment]
In the [current rootstock setup](https://github.com/manubot/rootstock/blob/2191d929a12a233347ee79c673b8f2971c10e6dd/SETUP.md), the following command is given:

```
git push --set-upstream origin main
```

Notice `main` instead of `master`. Perhaps your setup instructions are for an older version of rootstock?

**Comment by sebastiansukstorf** [comment]
I willow this instruction link https://github.com/manubot/rootstock/blob/master/SETUP.md from manubot.org

Mit freundlichem Gruß 
Sebastian Sukstorf

**Comment by dhimmel** [comment]
> I willow this instruction link https://github.com/manubot/rootstock/blob/master/SETUP.md from manubot.org

Ah our bad. We switched the default branch from `master` to `main` in https://github.com/manubot/rootstock/pull/399, but we forgot to update the website. I just updated the website in https://github.com/manubot/manubot.org/commit/9918b53a1ce7a25c6cd502aa175c20d87eea171f.

I would suggest starting the setup over with the instructions from https://github.com/manubot/rootstock/blob/main/SETUP.md. Sorry about the outdated link.

@agitter one thing we can consider is deleting the master branch. This would cause all links to the master branch to redirect to the current default branch (i.e. main). But I'm not sure if that would cause other unintended consequences for repos still using master.

**Comment by agitter** [comment]
@sebastiansukstorf I wanted to check in and see whether you were able to get this working.  If not, you could temporarily give me access to your private repository so I can help fix any setup issues.

**Comment by sebastiansukstorf** [comment]
Hello Anthony, no it's still not working. I grant you Access to the repo. Thank you. Sebastian

**Comment by agitter** [comment]
Thanks @sebastiansukstorf I received access.  I updated with the latest rootstock changes and pushed those so that you won't have to update again anytime soon.  This included back end changes to how the plugins work.  I also finished the setup steps but left `content/02.delete-me.md` for now.  Everything is building correctly now.

The failed build (run 491938772 from 27 days ago) that gave "Process completed with exit code 83." is related to the spellcheck.  The error that triggered this exit code was `Error: No word lists can be found for the language "de_DE".`  When you set `lang: en-EN` in the metadata, that fixed the builds.

The spellcheck does not currently support any languages besides English.  There were some barriers to general language support due how different tools that Manubot relies on use different language codes.  However, if you're interested in spellcheck, I could try to set it up as a special case.  If you're not going to use spellcheck, you can instead set `SPELLCHECK: false` in line 17 of `.github/workflows/manubot.yaml`.

Your manuscript should be ready for editing.  If you are unable to pull the changes I made, we could help clean up your local git clone.  Or the easier option could be to delete your local clone and clone a fresh copy of the repository from GitHub.

**Comment by sebastiansukstorf** [comment]
Hello Anthony, thank you very much! I appreciate your effort very much. I will check in the next days.
